move_period_from_transaction_date/model/account_bank_statement_line.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+"""Extend account.bank.statement.line to use transaction date in moves."""
+##############################################################################
+#
+#    Copyright (C) 2015 Therp BV - http://therp.nl.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, api
+
+
+class AccountBankStatementLine(models.Model):
+    """Extend account.bank.statement.line to use transaction date in moves."""
+    _inherit = 'account.bank.statement.line'
+
+    @api.one
+    def process_reconciliation(self, mv_line_dicts):
+        """ Retrieve the period derived from the statement line and store in
+        the context for further use """
+        periods = self.env['account.period'].find(dt=self.date)
```

Comment:
That is the default in 8.0: https://github.com/odoo/odoo/blob/8.0/addons/account/account.py#L1040